schemas/common/descriptors/relationshipdescriptor.schema.json

```diff
@@ -0,0 +1,54 @@
+{
+  "meta:license": [
+    "Copyright 2018 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/common/descriptors/relationshipdescriptor",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Relationship Descriptor",
+  "meta:extensible": true,
+  "meta:extends": [
+    "https://ns.adobe.com/xdm/common/desciptors/schemadescriptor#/definitions/descriptor"
+  ],
+  "meta:abstract": false,
+  "type": "object",
+  "description":
+    "A descriptor that describes a relationship between two schemas, a 'source' and a 'destination'.  This relationship may be directly between the schemas, or between specific properties of the schemas.  Relationship descriptors may be used to describe foreign key relationships between schemas, defining a relational model or star schema.",
+  "definitions": {
+    "relationshipdescriptor": {
+      "properties": {
+        "xdm:destinationSchema": {
+          "title": "Destination Schema",
+          "type": "string",
+          "format": "uri",
+          "description": "The destination schema this descriptor applies to"
+        },
+        "xdm:destinationProperty": {
+          "title": "Destination Property",
+          "type": "string",
+          "description":
+            "When present, the property of the destination schema to which this descriptor applies. This value is a JSON Pointer, applied to an instance of an object described by 'xdm:destSchema'"
+        },
+        "xdm:destinationItem": {
+          "title": "Destination Item",
+          "description":
+            "When present, the selector used to match a specific item in the array pointed to by destProperty",
+          "$ref":
+            "https://ns.adobe.com/xdm/common/descriptors/itemselector#/definitions/selector"
+        }
+      },
+      "required": ["xdm:destinationSchema"]
```

Comment:
It'll be good for the source schema related properties to also be part of relationshipDescriptor schema.json rather than in schemadescriptor.schema.json

Is there a reason for splitting source schema/destination schema properties across schemadescriptor.schema.json and relationshipDescriptor.schema.json?

Reply:
SchemaDescriptor is the base for RelationshipDescriptor. As such it is inheriting the source properties from that schema definition. We try to encourage reuse as much as possible in XDM so this is the right approach.

CC: @kstreeter